### PR TITLE
Fixes 8428: make it possible to choice a sampling method type when we create profile ingestion for the Snowflake

### PR DESCRIPTION
**Pull request by nakaken-churadata**
### Describe your changes:

Fixes 8428

I worked on that we can choice a sampling method type when we create a profiler ingestion for Snowflake because the current sampling method type, i.e `BERNOULLI`, occurs full table scan even though we use small sampling rate.
I created new test code for adding `sampling method type`.

By the way, I wrote some description in this PR but I'm not at good for English. I feel happy if someone review my writing :D

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [X] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [X] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by nakaken-churadata** [comment]
use `BERNOULLI`
image

use `SYSTEM`
image

**Comment by nakaken-churadata** [comment]
@sureshms What do I do for the result avobe?

File: ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py [hunk 1]
```diff
@@ -30,6 +33,11 @@ class SnowflakeSampler(SQASampler):
     """
 
     def __init__(self, *args, **kwargs) -> None:
+        self.sampling_method_type = SamplingMethodType.BERNOULLI
+
+        _profile_sample_config = kwargs.get("profile_sample_config", None)
+        if _profile_sample_config and _profile_sample_config.sampling_method_type:
+            self.sampling_method_type = _profile_sample_config.sampling_method_type
```
Comment (TeddyCr) [inline ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py:40]: I believe you can simplify this with 

```
self.sampling_method_type = kwargs.get("profile_sample_config", SamplingMethodType.BERNOULLI)
```

Comment (nakaken-churadata) [reply ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py:40]: I think that the value of `profile_sample_config` is assigned to the `self.sampling_method_type` if `profile_sample_config` is exist in the `kwargs`. But I want to assign the value of `profile_sample_config. sampling_method_type`.

Comment (nakaken-churadata) [reply ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py:40]: I tried your idea. 

```python
class SnowflakeSampler(SQASampler):
    def __init__(self, *args, **kwargs) -> None:
        if profile_sample_config:
            self.sampling_method_type = (
                profile_sample_config.sampling_method_type
                or SamplingMethodType.BERNOULLI
            )
        super().__init__(*args, **kwargs)
```

Next, I executed the test code for this code, and the error below occurred.

```
    def __init__(self, *args, **kwargs) -> None:
>       if profile_sample_config:
E       NameError: name 'profile_sample_config' is not defined
```

I think it is impossible to use local variables defined in another class even though it was a superclass.

Comment (nakaken-churadata) [reply ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py:40]: On the other hand, we can define arguments specifically like `BigQuerySampler`. I wonder which is better.
https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/bigquery/sampler.py

**Comment by nakaken-churadata** [comment]
new UI
image

File: ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py (hunk 1 above)
Comment (nakaken-churadata) [reply ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py]: I fixed to specify init arguments to `profile_sample_config` directly.
How about it?

Comment (nakaken-churadata) [reply ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py]: The code below includes a problem that no value is applied if `profile_sample_config` is None.

```python
if  profile_sample_config:
    self.sampling_method_type = profile_sample_config.sampling_method_type or SamplingMethodType.BERNOULLI
```

I want to apply `BERNOULLI` to `sampling_method_type` as a default value.
That is why I have arranged your proposed code like below.

```python
self.sampling_method_type = SamplingMethodType.BERNOULLI
if profile_sample_config:
    self.sampling_method_type = profile_sample_config.sampling_method_type
```

Comment (TeddyCr) [reply ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/snowflake/sampler.py]: Thanks for the change. We'll merge it once the tests passed

**Review by TeddyCr — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Comment by nakaken-churadata** [comment]
close #8428

**Pull request merged**